From 3849ae4274a71c222d1358f6b8e9da7bb750d35b Mon Sep 17 00:00:00 2001
From: codeworkx <daniel.hillenbrand@codeworkx.de>
Date: Thu, 23 Aug 2018 10:13:05 +0200
Subject: [PATCH 2/2] selinux: add domain for Gallery

Gallery needs to read vendor files on treble enabled devices.

This is based on policies written by Eino-Ville Talvala <etalvala@google.com> for Google camera
app.

Change-Id: I0e591c9b2a54c781203d5c49e6d6d65c2f477eaf
---
 common/private/gallery_app.te | 21 +++++++++++++++++++++
 common/private/seapp_contexts |  1 +
 2 files changed, 22 insertions(+)
 create mode 100644 common/private/gallery_app.te

diff --git a/common/private/gallery_app.te b/common/private/gallery_app.te
new file mode 100644
index 0000000..e71fe25
--- /dev/null
+++ b/common/private/gallery_app.te
@@ -0,0 +1,21 @@
+type gallery_app, domain, coredomain;
+
+app_domain(gallery_app)
+net_domain(gallery_app)
+
+# Access standard system services
+allow gallery_app app_api_service:service_manager find;
+allow gallery_app audioserver_service:service_manager find;
+allow gallery_app cameraserver_service:service_manager find;
+allow gallery_app drmserver_service:service_manager find;
+allow gallery_app mediacodec_service:service_manager find;
+allow gallery_app mediaextractor_service:service_manager find;
+allow gallery_app mediaserver_service:service_manager find;
+allow gallery_app mediametrics_service:service_manager find;
+allow gallery_app nfc_service:service_manager find;
+allow gallery_app surfaceflinger_service:service_manager find;
+
+allow gallery_app hidl_token_hwservice:hwservice_manager find;
+
+# Allow to read and execute camera app modules
+allow gallery_app vendor_file:file { rx_file_perms };
diff --git a/common/private/seapp_contexts b/common/private/seapp_contexts
index 2260302..9538d60 100644
--- a/common/private/seapp_contexts
+++ b/common/private/seapp_contexts
@@ -1 +1,2 @@
+user=_app isPrivApp=true seinfo=platform name=com.android.gallery3d domain=gallery_app type=app_data_file levelFrom=user
 user=_app isPrivApp=true seinfo=platform name=org.lineageos.snap domain=snap_app type=app_data_file levelFrom=user
-- 
2.17.1

